python/feldera/rest/feldera_client.py

```diff
@@ -188,18 +197,28 @@ def get_pipeline_stats(self, name: str) -> dict:
 
         return resp
 
-    def start_pipeline(self, pipeline_name: str):
+    def start_pipeline(self, pipeline_name: str, timeout_s: Optional[float] = None):

```

Comment:
In my opinion, there should by default be generous timeouts (e.g., 300 seconds for starting), such that there is a guarantee the function will return. Only for compilation I think it could be either disabled or set very generously.